python/cudf/cudf/pandas/module_accelerator.py

```diff
@@ -503,20 +506,19 @@ def disabled(self):
         -------
         Context manager for disabling things
         """
-        with self._use_fast_lib_lock:
-            # Have to hold the lock to modify this variable since
-            # another thread might be reading it.
-            # Modification has to happen with the lock held for the
-            # duration, so if someone else has modified things, then
-            # we block trying to acquire the lock (hence it is safe to
-            # release the lock after modifying this value)
-            saved = self._use_fast_lib
-            self._use_fast_lib = False
+        self._use_fast_lib_bkp[threading.get_ident()].append(
+            # The first time that a thread is disabled, unless it was the main
+            # thread it won't have an entry in the dictionary, which implicitly
+            # means it is running in fast mode.
```

Comment:
Q: What does the length of `self._use_fast_lib_bkp[threading.get_ident()]` signify? Can it be more than 1? When?

Reply:
I _think_ is signifies the number of times the thread's state was modified.

Reply:
Ah I see what you're asking. I've been having internet outages today and I didn't realize my last commit didn't get pushed. See https://github.com/rapidsai/cudf/pull/17955/commits/3ad5f3044c4762f2c01f53c008d1d6017c64fea7, which updates the test to verify that the context manager still works if it is nested. That answers your question: it can be of length>1 if the context manager is entered multiple times. In practice the sequence will always be [True, False, False, ...] so we could also maintain a count of some sort instead of a list. I can try that and see if there's an elegant way to express that that is more obviously readable.

Reply:
OK I reimplemented with a counter. That should make things more obvious.